x-pack/plugins/uptime/public/pages/synthetics/step_detail_page.tsx

```diff
@@ -5,11 +5,132 @@
  * 2.0.
  */
 
-import React from 'react';
-import { useParams } from 'react-router-dom';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
 import { useTrackPageview } from '../../../../observability/public';
 import { useInitApp } from '../../hooks/use_init_app';
 import { StepDetailContainer } from '../../components/monitor/synthetics/step_detail/step_detail_container';
+import { journeySelector } from '../../state/selectors';
+import { JourneyState } from '../../state/reducers/journey';
+import { JourneyStep } from '../../../common/runtime_types/ping/synthetics';
+import { StepPageNavigation } from '../../components/monitor/synthetics/step_detail/step_page_nav';
+import { useUiSetting$ } from '../../../../../../src/plugins/kibana_react/public';
+import { StepPageTitleContent } from '../../components/monitor/synthetics/step_detail/step_page_title';
+import { getJourneySteps } from '../../state/actions/journey';
+
+export const useStepDetailPage = (): {
+  activeStep?: JourneyStep;
+  checkGroup: string;
+  hasNextStep: boolean;
+  hasPreviousStep: boolean;
+  journey?: JourneyState;
+  stepIndex: number;
+} => {
+  const dispatch = useDispatch();
+
+  const { checkGroupId: checkGroup, stepIndex: stepIndexString } = useParams<{
+    checkGroupId: string;
+    stepIndex: string;
+  }>();
+
+  console.log('check group', checkGroup, 'step index', stepIndexString, useParams());
```

Comment:
Think you meant to delete this.

Reply:
I apologize. I thought I had reviewed this myself before opening up but clearly did not!